shell/cmd/cosh/cosh.go

```diff
@@ -114,3 +117,22 @@ func Build(c *Config) error {
 	}
 	return shell.NewShell().TranspileFile(c.Input, c.Output)
 }
+
+// Transpile transpiles all .cosh files in current directory
+// into corresponding .go files, without any further additions.
+// Use this for a multi-file codebase that is only used as a
+// go compiled package, not in interactive mode.
+func Transpile(c *Config) error {
```

Comment:
It seems to me that you could very easily detect whether these `package main` etc wrappers are already present and then add them if necessary in one unified `Build` command. This distinction between `Build` and `Transpile` is insufficient and will almost certainly cause confusion later. We don't want this to be like PHP. Alternatively, if you really can't automatically do it, add a bool flag for the unified `Build` command that specifies the behavior. The names `Build` and `Transpile` have absolutely no meaningful distinction.

Reply:
build assumes one file, transpile iterates over all files in the directory.